galvasr2/deduplicate.py

```diff
@@ -0,0 +1,123 @@
+import logging
+from galvasr2.align.spark.align_lib import load_audio_id_text_id_mapping, load_transcripts
+from datasketch import MinHash, MinHashLSH, MinHashLSHForest
+from nltk import ngrams
+from tqdm import tqdm
+import numpy as np
+import itertools
+
+
+class DataDeduplication:
+
+    def __init__(self, num_rows: int = 1):
+        self.num_rows = num_rows
+
+    def read_transcription_data(
+        self,
+        spark: SparkSession,
+        data_trans_index: str,
+        data_trans: str,
+    ) -> pd.DataFrame:
+        """Read the transcriptions
+        Returns
+        -------
+        transcripts_pdf: dataframe
+            pandas dataframe with the transcriptions
+        """
+        # spark.sparkContext.setLogLevel("INFO") # "ALL" for very verbose logging
+        logging.getLogger("py4j").setLevel(logging.ERROR)
+        catalogue_df = load_audio_id_text_id_mapping(spark, data_trans_index)
+        training_sample_rows = catalogue_df.collect()
+        # Comment this out to load everything. It might takes ~15 minute, in my experience, on an 8 core machine.
+        if self.num_rows > 1:
```

Comment:
I am not enthusiastic about self.num_rows == 1 being a special case. I would recommend declaring `num_rows: Option[int] = None` in `__init__` instead. Then you can do `if self.num_rows is not None:` as the condition here.